bakery/src/scripts/link-book.py

```diff
@@ -0,0 +1,61 @@
+"""
+Replaces legacy module ids in links to external modules with
+uuids from the target module and corresponding canonical book.
+"""
+
+import sys
+import os
+import json
+from lxml import etree
+import requests
+
+
+def main():
+    data_dir, canonical_list = sys.argv[1:3]
+
+    # define the canonical books
+    with open(canonical_list) as canonical_file:
+        canonical_books = json.load(canonical_file)["canonical_books"]
+        canonical_ids = [book["uuid"] for book in canonical_books]
+
+    # load assembled collection
+    assembled_collection = f"{data_dir}/collection.assembled.xhtml"
+    doc = etree.parse(assembled_collection)
+
+    # look up uuids for external module links
+    for node in doc.xpath(
+        '//x:a[@href and starts-with(@href, "/contents/m")]',
+        namespaces={"x": "http://www.w3.org/1999/xhtml"},
+    ):
+        # find legacy module id
+        link = node.attrib["href"]
+        module_id = link.lstrip("/contents/")
+
+        # get target module uuid
+        req = requests.get(f"https://archive.cnx.org/content/{module_id}")
+        if req.status_code != 200:
+            continue
```

Comment:
Instead of masking potential errors, should we just `req.raise_for_status()` and let it error out? (the idea being it's better to fail than lead to non-deterministic builds) Or am I misunderstanding something?